lang/csharp/src/apache/main/IO/JsonDecoder.cs

```diff
@@ -45,6 +45,14 @@ private class ReorderBuffer
             public JsonReader OrigParser { get; set; }
         }
 
+        private sealed class AvroJsonTextReader : JsonTextReader
+        {
+            public AvroJsonTextReader(TextReader reader) : base(reader)
```

Comment:
`public` ctor for a `private` class. I am not a C# dev but it usually the visibility of the ctor should be local, i.e. `private` too

Reply:
Hey @martin-g a `public` ctor is needed when initializing the `reader` in the `Configure` methods. Having it `private` would cause a compiler error

Reply:
So, the class is private but visible. The constructor is not ...
Oh, Microsoft ...